src/main/java/gregtech/common/items/MetaItem1.java

```diff
@@ -477,52 +659,127 @@ public void registerRecipes() {
             .outputs(BATTERY_RE_HV_SODIUM.getStackForm())
             .duration(1600).EUt(2)
             .buildAndRegister();
-    }
 
-    @Override
-    public boolean isBeaconPayment(ItemStack stack) {
-        int damage = stack.getMetadata();
-        if (damage >= this.metaItemOffset) {
-            return false;
-        }
-        Material material = Material.MATERIAL_REGISTRY.getObjectById(damage % 1000);
-        OrePrefix prefix = this.orePrefixes[(damage / 1000)];
-        if(prefix != null && material != null) {
-            boolean isSolidState = prefix == OrePrefix.ingot || prefix == OrePrefix.gem;
-            boolean isMaterialTiered = material instanceof SolidMaterial && ((SolidMaterial) material).harvestLevel >= 2;
-            return isSolidState && isMaterialTiered;
-        }
-        return false;
-    }
+        // Dyes recipes
```

Comment:
Please move all those recipes to the recipe classes.

Reply:
i think @DStrand1 wants to do a big refractor so ill wait for his input